antismash/common/all_orfs.py

```diff
@@ -89,26 +89,28 @@ def scan_orfs(seq: str, direction: int, offset: int = 0) -> List[FeatureLocation
     return sorted(matches, key=lambda x: min(x.start, x.end))
 
 
-def create_feature_from_location(record: Record, location: FeatureLocation,
-                                 counter: int = 1, label: Optional[str] = None) -> CDSFeature:
+def create_feature_from_location(record: Record, location: FeatureLocation, name: str) -> CDSFeature:
     """ Creates a CDS feature covering the provided location.
 
         Arguments:
             record: The Record the CDSFeature will belong to, used to generate
                     the feature translation
             location: The FeatureLocation specifying the location of the CDSFeature
-            counter: An integer to use to format a default label 'allorf' with,
-                     used only if label not provided
-            label: The locus tag, protein id, and gene name to use for the new
-                   CDSFeature
+            name: The locus tag, protein id, and gene name to use for the new
+                  CDSFeature
 
         Returns:
             The CDSFeature created.
     """
-    if label is None:
-        label = 'allorf%03d' % counter
+    if not isinstance(record, Record):
```

Comment:
This isn't really needed, since we're using methods of the record in this function anyway. Type checking is more important where the value is stored and used at some arbitrary later point (like many of the property setters throughout secmet). Value checks are still important though.

This function doesn't add the CDS to the record anyway, so it's a little strange to mention that in the error message.